data/mp/multiplay/skirmish/rules.js

```diff
@@ -422,25 +379,26 @@ function eventGameInit()
 		else // CAMP_WALLS
 		{
 			setPower(2500, playnum);
-			for (var count = 0; count < techlist.length; count++)
-			{
-				completeResearch(techlist[count], playnum);
-			}
+			completeResearchOnTime (timeAdvancedBaseTech, playnum);
 		}
-	}
+	
 
-	var techLevel = getMultiTechLevel();
-	if (techLevel == 2)
-	{
-		grantTech(TECH_TWO);
-	}
-	else if (techLevel == 3)
-	{
-		grantTech(TECH_THREE);
-	}
-	else if (techLevel == 4)
-	{
-		grantAllTech();
+		var techLevel = getMultiTechLevel();
+		if (techLevel == 2)
+		{
+			completeResearchOnTime (timeT2, playnum);
+		
+
+		}
+		else if (techLevel == 3)
+		{
+			completeResearchOnTime (timeT3, playnum);
+		
+		}
+		else if (techLevel == 4)
+		{
+			completeResearchOnTime (60*60*24, playnum);
```

Comment:
You should be able to use [Infinity](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Infinity) here for the time since everything will be less than it.

Reply:
nice, I did not know about the current convenient way.
Thought of the need for workarounds.